tests/testthat/test-test-threads.R

```diff
@@ -20,25 +20,33 @@ cleanup_bbi <- function(.recreate_dir = FALSE) {
 cleanup_bbi(.recreate_dir = TRUE)
 
 
-get_est_options <- function(search_str = "MAXEVAL|NITER|NBURN", mods){
+get_est_options <- function(mods, search_opts = c("maxeval", "niter", "nburn")){
 
-  max_evals <- map(mods, function(.mod){
+  mod_names <- purrr::map_chr(mods, ~ basename(.x$absolute_model_path))
+  est_opts <- map(mods, function(.mod){
     mod_path <- get_model_path(.mod)
     mod_lines <- mod_path %>% readLines()
-    str_line_loc <- which(grepl(search_str, mod_lines))
-    str_values <- str_split(mod_lines[str_line_loc], " ")
-
-    values <- map(1:length(str_values), ~ {
-      est_loc <- grepl("METHOD", str_values[[.x]])
-      est_method <- str_values[[.x]][est_loc]
-      str_loc <- grepl(search_str, str_values[[.x]])
-      eval_method <- gsub('[[:digit:]]+|=', '', str_values[[.x]][str_loc])
-      value <- as.numeric(gsub(glue('{search_str}|='), '', str_values[[.x]][str_loc]))
-      names(value) = paste0(est_method, ", ",eval_method)
-      value
-    }) %>% unlist()
+    ctl <- nmrec::parse_ctl(mod_lines)
```

Comment:
Okay as is, but `mod_lines` is only used in the `parse_ctl` call, so you could just use `read_ctl` directly.